src/main/java/com/rabbitmq/jms/client/RMQSession.java

```diff
@@ -425,6 +435,7 @@ public void commit() throws JMSException {
                 // Call commit on the channel.
                 // All messages ought already to have been acked.
                 this.channel.txCommit();
+                this.uncommittedMessageTags.clear();
```

Comment:
I'd place the clearing of the uncommitted tags in a dedicated method that would check the state of the flag and do nothing when the feature is disabled. That would save a call on a synchronized data structure for those who don't use it.